core/domain/classifier_services.py

```diff
@@ -100,8 +101,38 @@ def classify_string_classifier_rule(state, normalized_answer):
             training_examples.extend([
                 [doc, [str(answer_group_index)]]
                 for doc in classifier_rule_spec.inputs['training_data']])
-    if len(training_examples) > 0:
+
+    exploration = exp_services.get_exploration_by_id(exp_id)
+    state_obj = exploration.states[state_name]
+    algorithm_id = feconf.INTERACTION_CLASSIFIER_MAPPING['TextInput']
+    if state_obj.classifier_model_id:
+        classifier = get_classifier_by_id(state_obj.classifier_model_id)
+        sc_dict = classifier.cached_classifier_data
+        sc = classifier_registry.Registry.get_classifier_by_algorithm_id(
+            algorithm_id)
+        sc.from_dict(sc_dict)
```

Comment:
Oh, yep it was tested. When you execute classifier_services_test, there are a total of four calls to the function classify(). The first call goes into the 'else' branch and the remaining three calls go into the "if" branch. This is how the cached classifier is being used, right? Should we have separate tests checking that?